druntime/src/core/thread/osthread.d

```diff
@@ -2190,6 +2194,29 @@ else version (Posix)
         __gshared sem_t suspendCount;
 
 
+        extern (C) void thread_preSuspend( void* sp ) nothrow {
+            // NOTE: Since registers are being pushed and popped from the
+            //       stack, any other stack data used by this function should
+            //       be gone before the stack cleanup code is called below.
+            Thread obj = Thread.getThis();
+            assert(obj !is null);
+
+            if ( !obj.m_lock )
+            {
+                obj.m_curr.tstack = getStackTop();
+            }
+        }
+
+        extern (C) void thread_postSuspend() nothrow {
+            Thread obj = Thread.getThis();
+            assert(obj !is null);
+
+            if ( !obj.m_lock )
+            {
+                obj.m_curr.tstack = obj.m_curr.bstack;
```

Comment:
I honestly don't understand why this one is necessary, it would warrant a comment from someone who knows, but as things go, I kept it as this.